dcos_test_utils/marathon.py

```diff
@@ -424,3 +396,25 @@ def deploy_and_cleanup(self, app_definition, timeout=120, check_health=True, ign
     def deploy_pod_and_cleanup(self, pod_definition, timeout=300):
         yield self.deploy_pod(pod_definition, timeout=timeout)
         self.destroy_pod(pod_definition['id'], timeout)
+
+    def purge(self):
+        """ Deletes all applications, all pods, and then waits
+        indefinitely for any deployments to finish
+        """
+        for app in self.get('v2/apps').json()['apps']:
+            log.info('Purging application: {}'.format(app['id']))
+            self.delete('v2/apps' + app['id'])
```

Comment:
there is a trick in marathon where you can delete all apps by deleting the root group
https://github.com/mesosphere/marathon/blob/master/tests/scale/common.py#L82
this allows you do delete all apps without getting the list or cycling through them.

this doesn't work for pods :(